ftwhttp/request.go

```diff
@@ -102,9 +104,23 @@ func (r *Request) AddHeader(name string, value string) {
 
 // AddStandardHeaders adds standard headers to the request, if they don't exist
 //
-// This will add Content-Length and the proper Content-Type
-func (r *Request) AddStandardHeaders(size int) {
-	r.headers.AddStandard(size)
+// AddStandardHeaders does the following:
+//   - adds `Connection` header with `close` value (if not set) to improve performance
+//   - adds `Content-Length` header if payload size > 0 or the request method
+//     permits a body (the spec says that the client SHOULD send `Content-Length`
+//     in that case)
+func (r *Request) AddStandardHeaders() {
+	if r.headers.Get("Connection") == "" {
+		r.headers.Add("Connection", "close")
+	}
+
+	if len(r.data) > 0 || methodsWithBodyRegex().MatchString(r.requestLine.Method) {
+		r.headers.Add("Content-Length", strconv.Itoa(len(r.data)))
+	}
+}
+
+func methodsWithBodyRegex() *regexp.Regexp {
+	return regexp.MustCompile(`^POST|PUT|PATCH|DELETE$`)
```

Comment:
I'd rather have this defined as a constant at file scope. It's not a huge deal but this regex will be applied many times during a full test run. Thus I'd like to cache compilation.

```go
var methodsWithBodyRegex = regexp.MustCompile(`^POST|PUT|PATCH|DELETE$`)
```

Reply:
I see! Good catch. 👍

Reply:
@theseion Updated https://github.com/coreruleset/go-ftw/pull/135/commits/4bfdaca220be250a5e00860177778608ace60aad